pkg/appgw/frontend_listeners.go

```diff
@@ -66,23 +65,24 @@ func (c *appGwConfigBuilder) getListenerConfigs(ingressList []*v1beta1.Ingress)
 func (c *appGwConfigBuilder) newListener(listener listenerIdentifier, protocol n.ApplicationGatewayProtocol, envVariables environment.EnvVariables) n.ApplicationGatewayHTTPListener {
 	frontendPortName := generateFrontendPortName(listener.FrontendPort)
 	frontendPortID := c.appGwIdentifier.frontendPortID(frontendPortName)
+	usePrivateIP := listener.UsePrivateIP || envVariables.UsePrivateIP == "true"
 
 	return n.ApplicationGatewayHTTPListener{
 		Etag: to.StringPtr("*"),
 		Name: to.StringPtr(generateListenerName(listener)),
 		ApplicationGatewayHTTPListenerPropertiesFormat: &n.ApplicationGatewayHTTPListenerPropertiesFormat{
 			// TODO: expose this to external configuration
-			FrontendIPConfiguration: resourceRef(*c.getIPConfigurationID(envVariables)),
+			FrontendIPConfiguration: resourceRef(*GetIPConfigurationID(&c.appGw, usePrivateIP)),
 			FrontendPort:            resourceRef(frontendPortID),
 			Protocol:                protocol,
 			HostName:                &listener.HostName,
 		},
 	}
 }
 
-func (c *appGwConfigBuilder) getIPConfigurationID(envVariables environment.EnvVariables) *string {
-	usePrivateIP, _ := strconv.ParseBool(envVariables.UsePrivateIP)
-	for _, ip := range *c.appGw.FrontendIPConfigurations {
+// GetIPConfigurationID returns ID if given IP exists
+func GetIPConfigurationID(appGw *n.ApplicationGateway, usePrivateIP bool) *string {
```

Comment:
It might be good to add a test for this - just to make sure we don't break it accidentally